Specify the 'downloads' directory

golirev opened the pull request:
I added a way to specify the folder to place downloaded files.

You can share a "downloads" folder with multiple windows installer build instances
without copying from one instance to another.

  rake DOWNLOADS="c:\downloads" # specify the folder to place downloaded files, absolute/relative

In addition above, I added the ability to handle dependency files 
more properly in multi-toolchain.

Download a set of dependency files from distinguished URL path
and store them under the "downloads" sub-directory separately.

config/compilers/mingw64.rb is just a sample.
You'll fail in downloading the dependency files, since there aren't such files.

:knap_path value is used for separation.

golirev committed:
Specify the 'downloads' directory
And deal multi-toolchain dependent files properly

golirev committed:
Specify the 'downloads' directory
And deal multi-toolchain dependent files properly

Azolo commented on recipes/dependencies/knapsack.rake:
```diff
@@ -13,19 +13,33 @@ dependencies.each do |dependency_key, dependency|
+      if compiler.knap_path
+        download_path = "#{download_path}/#{compiler.knap_path}"
+      end
 
```
Could you clean up the tabs in this file?

golirev replied:
I modified the indent.
Thank you.

Azolo commented on config/compilers/mingw64.rb:
```diff
@@ -1,5 +1,151 @@
+        ],
+      }
+    )
+
```
I'm glad that someone is doing this kind of stuff. =)

Azolo commented:
It looks promising!

golirev committed:
Modify the indent

golirev committed:
Modify the indent

Azolo commented:
I'm in the middle of getting some other stuff together, so give me a couple days and I'll look at it more closely. =)

vais committed:
Upgrade Ruby 2.2 and 2.3 to 2.2.6 and 2.3.3 respectively

Azolo committed:
Change History to reflect unreleased state

Azolo committed:
Merge branch 'vais-master'

Azolo committed:
Update Dependencies

- OpenSSL, libffi, LibYAML, tcl, tk

Azolo committed:
Retry timeouts with SourceForge

* SourceForge has become unreliable lately

Azolo committed:
Prep for release

- 2.2.6 `make test-all`: https://gist.github.com/Azolo/b9a2a55ff2e94265de882ee7aec9c663
- 2.3.3 `make test-all`: https://gist.github.com/Azolo/6f661dfceddd1ef8580bfde0c79eccb1

golirev committed:
Merge branch 'gl-multi-toolchain' of https://github.com/golirev/rubyinstaller into gl-multi-toolchain

ghost commented:
may need updating the software versions.
